File: GO/services/services.go
package services

import(
	"strings"
	"../merge"
	"../bitcode"
	"../graph"
)

func SortEmails(filename string) string{
	name := strings.Split(filename,".")[0]
	merge.FilterFile(name+"/"+filename,"(\\w[-._\\w]*\\w@\\w[-._\\w]*\\w\\.\\w{2,3})")
	merge.CreateLeaves(name+"/"+filename+".filtered",5)
	merge.MergeSort(name+"/leaves/")

	return merge.GetSortedFile(filename)
}

func HideMessage(filename, message string) string {
	bitcode.HideMessage(filename,message)
	return bitcode.GetHidden(filename)
}

func SeekMessage(filename string) string {
	bitcode.SeekMessage(filename)
	return bitcode.GetMessage(filename)
}

func Kruskal(json_graph *graph.Graph) string {
	// new_g := graph.JsonToGraph([]byte(json_graph))
	// log.Printf("json_graph:\n%v",new_g)
	tree := graph.ApplyKruskal(*json_graph)
	return graph.GraphToJson(*tree)
}

File: Ruby/services/services.rb
require 'json'
require_relative '../merge/merge'
require_relative '../bitcode/bitcode'
require_relative '../graph/kruskal'

def SortEmails(filename)
	puts filename
	name = "#{filename.split(".")[0]}"
	FilterFile("#{name}/#{filename}",/\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i)
	CreateLeaves("#{name}/#{filename}.filtered",5)
	MergeSort("#{name}/leaves/")

	GetSortedFile(filename)
end

def HideMessage(filename, message)
	hideMessage(filename,message)
	GetHidden(filename)
end

def SeekMessage(filename)
	seekMessage(filename)
	GetMessage(filename)
end

def Kruskal(json_graph)
	tree = ApplyKruskal(json_graph)
	tree.to_json
end

# SortEmails("2milmails.txt")

# HideMessage("oceano.bmp","que pedo")

# SeekMessage("hidden_oceano.bmp")

# f = File.open("graph/graph.json",'r')
# data = f.read

# puts Kruskal(JSON.parse(data))